app/editPlan/getTaskListData.js

```diff
@@ -0,0 +1,107 @@
+  else if (statusCount.length >= 2) {
+    statusText = COMPLETED
+    complete = true
+    hrefText = REVIEW_DETAILS
+  }
+  return { statusText, complete, hrefText }
+}
+
+const getAboutTheIndividual = ({ comments }, stub) => {
+  return {
+    heading: {
+      text: 'Write about the individual',
+    },
+    items: [
+      {
+        text: 'Responsivity',
+        href: `${stub}/review-responsivity`,
+        ...reviewComments(comments, ['YOUR_RESPONSIVITY', 'THEIR_RESPONSIVITY']),
+      },
+    ],
+  }
+}
+const actionsComplete = (
+  complete,
+  { description = '', intervention = '', motivationUUID = '', updated = '', owner = [], ownerOther = '' }
+) =>
+  complete &&
+  (description.length > 0 || intervention.length > 0) &&
+  motivationUUID.length > 0 &&
+  updated.length > 0 &&
+  (owner.length > 0 || ownerOther.length > 0)
+const getAddObjectives = ({ objectives = [] }, stub) => {
+  const addedObjectives = objectives.map(({ actions = [], needs = [], description = '', id }, index) => {
+    const addedObjective = {
+      text: description.length > 0 ? description : `Add objective ${index + 1}`,
+      hrefText: ADD_DETAILS,
+      href: `${stub}/edit-objective/${id}`,
+      statusText: INCOMPLETE,
+      complete: false,
+    }
+    if (description.length === 0 && needs.length === 0) {
+      addedObjective.statusText = NOT_STARTED
+    } else if (description.length > 0 && needs.length > 0 && actions.reduce(actionsComplete, true)) {
+      addedObjective.statusText = COMPLETED
+      addedObjective.complete = true
+    }
+    return addedObjective
+  })
+
+  const addAnotherObjective =
+    addedObjectives.length === 1 && addedObjectives[addedObjectives.length - 1].statusText === NOT_STARTED
+      ? {
+          text: `Add objective 2`,
+          hrefText: 'Add objective 1 first',
+          statusText: CANNOT_START_YET,
+          complete: false,
+        }
+      : {
+          text: `Add objective ${addedObjectives.length + 1}`,
+          hrefText: ADD_DETAILS,
+          href: `${stub}/edit-objective/NEW`,
+          statusText: NOT_STARTED,
+          complete: false,
+        }
+
+  return {
+    heading: {
+      text: 'Add objectives',
+      description: 'Each objective can have multiple actions.',
+    },
+    items: [...addedObjectives, addAnotherObjective],
+  }
+}
+const getFinalInformation = ({ comments }, stub, hasObjective) => {
+  const itemOptions = hasObjective
+    ? reviewComments(comments, ['YOUR_SUMMARY', 'THEIR_SUMMARY'])
+    : { hrefText: 'Add at least one objective first', statusText: CANNOT_START_YET, complete: false }
+  return {
+    heading: {
+      text: 'Enter final information',
+    },
+    items: [
+      {
+        text: 'Decisions and comments',
+        href: hasObjective ? `${stub}/review-responsivity` : '',
+        ...itemOptions,
+      },
+    ],
+  }
+}
+
+module.exports = { getAboutTheIndividual, getAddObjectives, getFinalInformation }
```

Comment:
I think there is scope to utilise express-validator for the status checking above. I don't think its an exact fit, but I think it might be worth raising a ticket to address.

Reply:
raised SP-481 Investigate using express validator for validating completion status of plan elements (https://dsdmoj.atlassian.net/browse/SP-481)